lib/rules/no-insecure-url.js

```diff
@@ -88,15 +109,24 @@ module.exports = {
                     const rawStringText = node.value.raw;
                     const cookedStringText = node.value.cooked;
 
-                    if ((matches(blocklist, rawStringText) && !matches(exceptions, rawStringText)) ||
-                        (matches(blocklist, cookedStringText) && !matches(exceptions, cookedStringText))) {
-                        context.report({
-                            node: node,
-                            messageId: "doNotUseInsecureUrl"
-                        });
-                    }
-                }
-            }
+                  if (shouldFix(context, node) && (matches(blocklist, rawStringText) && !matches(exceptions, rawStringText)) ||
+                      (matches(blocklist, cookedStringText) && !matches(exceptions, cookedStringText))) {
+                      context.report({
+                        node: node,
+                        messageId: "doNotUseInsecureUrl",
+                        fix(fixer) {
+                          // Only fix if it contains an http url
+                          if (node.value.raw.toLowerCase().includes("http")) {
+                            let escapedString =  JSON.stringify(context.getSourceCode().getText(node)).replaceAll('"', ""); // delete just last and first
+                            let fixedString = escapedString.replace(/http:/gi, "https:");
+                            //insert an "s" before ":/" to change http:/ to https:/
+                            return fixer.replaceText(node, fixedString);

```

Comment:
Why the regex?
Didn't we learn in https://github.com/eslint/eslint/issues/16061#issuecomment-1167484065 that we can use something like `context.getSourceCode().getText(node).replace("http://", "https://")`?

Reply:
Right, 
Are you referring to line 121 or to 120? 
line 120 is to escape the string which we need for the case that we have something like 
```javascript
var url = `http://moz\ti\tlla.org\`;
```
I see that this line looks a bit hacky. I will rewrite it.


In line 121 we are replacing all occurrences of "http:" to "https:" to ensure that for example a  query in a line like 
```javascript
var url = `http://example.com?redirectAfterLoadTo=http://...;`
```
gets upgraded. Might be an edge case so we could also remove the `\g` since it also leads to incorrect fixes if the second URL is e.g. exempted.